packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+        snapshotCacheExpiryMs: 0,
+        inactiveTimeoutMs,
+    };
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                maxAckWaitTime: 10000,
+                maxOpsSinceLastSummary: 7000,
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+            },
+        },
+        gcOptions,
+    };
+
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const settings = {
+        "Fluid.GarbageCollection.Test.Tombstone": "true",
+        "Fluid.GarbageCollection.TestOverride.SweepTimeoutMs": sweepTimeoutMs,
+    };
+    const configProvider = mockConfigProvider(settings);
+    const createContainer = async () => provider.createContainer(runtimeFactory, { configProvider });
+    const loadSummarizerAndContainer = async (summaryVersion?: string) => {
+        const absoluteUrl = await mainContainer.getAbsoluteUrl("");
+        return createSummarizerWithContainer(provider, absoluteUrl, runtimeFactory, { configProvider }, summaryVersion);
+    };
+    const summarize = async (summarizer: ISummarizer) => {
+        await provider.ensureSynchronized();
+        return summarizeNow(summarizer);
+    };
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+        mainContainer = await createContainer();
+        mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    const createUnreferencedDataStore = async () => {
+        const handleKey = "handle";
+        const testDataObject = await dataObjectFactory.createInstance(mainDataStore.containerRuntime);
+        const testDataObjectId = testDataObject.id;
+
+        // Reference a datastore and summarize
+        mainDataStore._root.set(handleKey, testDataObject.handle);
+        const {
+            container: summarizingContainer1,
+            summarizer: summarizer1,
+        } = await loadSummarizerAndContainer();
+        await summarize(summarizer1);
+
+        // Unreference a datastore and summarize
+        mainDataStore._root.delete(handleKey);
+        const summaryVersion = (await summarize(summarizer1)).summaryVersion;
+        testDataObject._root.set("send while unreferenced", "op");
```

Comment:
So... the resulting op from this is fine here because SweepTimeout hasn't elapsed, but will be a trailing op when a future container loads after SweepTimeout, and will trip on the Tombstone? Did I get that right?

Reply:
So Tombstone doesn't get tripped up for a few different reasons.
- Before we actually `getGCData`, we `process` all trailing ops in the non-tombstone state. 
  - Then we flag the datastoreContext as tombstone
- The `process` call that we use to `process` any ops during summarization doesn't go through the tombstoning logic.

Reply:
No GC logic - initializing base state and collecting garbage runs after the trailing ops are processed. Since the ops were fine to send at the time they were sent, they're fine to send now.

Reply:
Here's a task for testing for trailing ops https://dev.azure.com/fluidframework/internal/_workitems/edit/2320